development/scripts/fix-spec.py

```diff
@@ -0,0 +1,107 @@
+                supply_property = schema['properties']['supply']
+                if 'properties' in supply_property and 'value' in supply_property['properties']:
+                    value_property = supply_property['properties']['value']
+                    if 'enum' in value_property and set(value_property['enum']) == {'ac', 'dc'}:
+                        print(f"Replacing complex 'supply' field in PowerFeed")
+                        schema['properties']['supply'] = {
+                            'type': 'string',
+                            'enum': ['ac', 'dc'],
+                            'default': 'ac'
+                        }
+
+            # Replace nested `phase` field in PowerFeed
+            if 'phase' in schema['properties']:
+                phase_property = schema['properties']['phase']
+                if 'properties' in phase_property and 'value' in phase_property['properties']:
+                    value_property = phase_property['properties']['value']
+                    if 'enum' in value_property and set(value_property['enum']) == {'single-phase', 'three-phase'}:
+                        print(f"Replacing complex 'phase' field in PowerFeed")
+                        schema['properties']['phase'] = {
+                            'type': 'string',
+                            'enum': ['single-phase', 'three-phase'],
+                            'default': 'single-phase'
+                        }
+        
+        # Handle Prefix schema
+        if name == 'Prefix' and 'properties' in schema:
+            # Replace complex `type` object with a simpler string enum in Prefix
+            if 'type' in schema['properties']:
+                type_property = schema['properties']['type']
+                if 'properties' in type_property and 'value' in type_property['properties']:
+                    print(f"Replacing complex 'type' field in Prefix")
+                    schema['properties']['type'] = {
+                        'type': 'string',
+                        'enum': ['container', 'network', 'pool'],
+                        'default': 'network'
+                    }
+                    
+        if 'properties' in schema:
+            # Remove "null" item from nullable enums
+            for name, prop in schema['properties'].items():
+                if 'enum' in prop and None in prop['enum']:
+                    prop['enum'].remove(None)
+                if 'properties' in prop and 'value' in prop['properties'] and 'enum' in prop['properties']['value'] and None in prop['properties']['value']['enum']:
+                    prop['properties']['value']['enum'].remove(None)
+
+            # Fix nullable types
+            nullable_types = [
+                'parent_device',
+                'primary_ip',
+            ]
+
+            for ntype in nullable_types:
+                if ntype in schema['properties']:
+                    schema['properties'][ntype]['nullable'] = True
```

Comment:
Looking at the schema, I'm seeing where `parent_device` already has `nullable: true` in the schema, and I'm not seeing any `primary_ip` properties (only `primary_ip4` and `primary_ip6`)

Reply:
This was also kept from the netbox script, will check tomorrow if neccessary like above.